### PR TITLE
update to 1.20.6

### PR DESCRIPTION
**Pull request by uku3lig**
title says it all !

### CODE REVIEW

**Review by TheKodeToad — APPROVED** [review]
nice, thanks!
I never got around to checking whether any changes were needed and nobody annoyed me on CurseForge xD

**Comment by TheKodeToad** [comment]
Oh, looks like this doesn't work quite properly
\- It doesn't affect the ping tooltip
\- When hovering the incompatible server text the game crashes due to it being a List<OrderedText> and the implicit cast due to generics...:
```
java.lang.ClassCastException: class net.minecraft.class_5481$$Lambda/0x00007f06c8dc5a30 cannot be cast to class net.minecraft.class_2561 (net.minecraft.class_5481$$Lambda/0x00007f06c8dc5a30 and net.minecraft.class_2561 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @52af6cff)
	at net.minecraft.class_4267$class_4270.args$znd000$numeralping$setTooltip(class_4267.java:613)
	at net.minecraft.class_4267$class_4270.method_25343(class_4267.java:320)
	at net.minecraft.class_350.method_44397(class_350.java:438)
	at net.minecraft.class_350.method_25311(class_350.java:423)
	at net.minecraft.class_350.method_48579(class_350.java:187)
	at net.minecraft.class_339.method_25394(class_339.java:66)
	at net.minecraft.class_437.method_25394(class_437.java:131)
	at net.minecraft.class_500.method_25394(class_500.java:250)
	at net.minecraft.class_437.method_47413(class_437.java:120)
	at net.minecraft.class_757.mixinextras$bridge$method_47413$180(class_757.java)
	at net.minecraft.class_757.wrapOperation$zmc000$fabric-screen-api-v1$onRenderScreen(class_757.java:2101)
	at net.minecraft.class_757.method_3192(class_757.java:922)
	at net.minecraft.class_310.method_1523(class_310.java:1320)
	at net.minecraft.class_310.method_1514(class_310.java:888)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
```

Would you like to have a go at fixing this? Or should I merge now and try to fix myself?

**Comment by uku3lig** [comment]
thanks for noticing that, i targeted the wrong method by accident but it should work just fine now (at least it does on my machine)

i replaced your @ModifyArgs with mixinextra's @WrapWithCondition, as it allows for slightly cleaner code, better compatibility and because setting the tooltip text to null caused another crash

File: src/main/java/io/toadlabs/numeralping/mixin/ServerEntryMixin.java [hunk 1]
```diff
@@ -35,12 +34,9 @@ public void shiftText(Args args) {
...
 	// hide the tooltip if it's redundant
-	@ModifyArgs(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setMultiplayerScreenTooltip(Ljava/util/List;)V"), require = 0)
-	public void setTooltip(Args args) {
-		var tooltip = (List<Text>) args.get(0);
-		if (tooltip != null && tooltip.size() == 1 && NumeralConfig.instance().serverList && tooltip.get(0).getContent() instanceof TranslatableTextContent translatable && translatable.getKey().equals("multiplayer.status.ping")) {
-			args.set(0, null);
-		}
+	@WrapWithCondition(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setTooltip(Lnet/minecraft/text/Text;)V"))
+	public boolean hideTooltip(MultiplayerScreen instance, Text text) {
+		return !NumeralConfig.instance().serverList;
```
Comment (TheKodeToad) [inline src/main/java/io/toadlabs/numeralping/mixin/ServerEntryMixin.java:39]: The `multiplayer.status.ping` was there for a good reason - only the ping is redundant but other information can be displayed

Comment (uku3lig) [reply src/main/java/io/toadlabs/numeralping/mixin/ServerEntryMixin.java]: i foror

**Pull request merged**